src/main/java/io/cryostat/net/web/http/api/beta/CredentialTestPostHandler.java

```diff
@@ -0,0 +1,177 @@
+import io.cryostat.net.web.http.api.v2.IntermediateResponse;
+import io.cryostat.net.web.http.api.v2.RequestParameters;
+
+import com.google.gson.Gson;
+import io.vertx.core.http.HttpMethod;
+import org.apache.commons.lang3.StringUtils;
+
+public class CredentialTestPostHandler extends AbstractV2RequestHandler<CredentialTestResult> {
+
+    static final String PATH = "credentials/:targetId";
+
+    private final TargetConnectionManager tcm;
+
+    @Inject
+    CredentialTestPostHandler(
+            AuthManager auth,
+            CredentialsManager credentialsManager,
+            Gson gson,
+            TargetConnectionManager tcm) {
+        super(auth, credentialsManager, gson);
+        this.tcm = tcm;
+    }
+
+    @Override
+    public boolean requiresAuthentication() {
+        return true;
+    }
+
+    @Override
+    public ApiVersion apiVersion() {
+        return ApiVersion.BETA;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.POST;
+    }
+
+    @Override
+    public Set<ResourceAction> resourceActions() {
+        return EnumSet.of(ResourceAction.READ_TARGET);
+    }
+
+    @Override
+    public String path() {
+        return basePath() + PATH;
+    }
+
+    @Override
+    public List<HttpMimeType> produces() {
+        return List.of(HttpMimeType.JSON);
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    public IntermediateResponse<CredentialTestResult> handle(RequestParameters params)
+            throws Exception {
+        String targetId = params.getPathParams().get("targetId");
+        String username = params.getFormAttributes().get("username");
+        String password = params.getFormAttributes().get("password");
+        if (StringUtils.isAnyBlank(targetId, username, password)) {
+            StringBuilder sb = new StringBuilder();
+            if (StringUtils.isBlank(targetId)) {
+                sb.append("\"targetId\" is required.");
+            }
+            if (StringUtils.isBlank(username)) {
+                sb.append("\"username\" is required.");
+            }
+            if (StringUtils.isBlank(password)) {
+                sb.append(" \"password\" is required.");
+            }
+
+            throw new ApiException(400, sb.toString().trim());
+        }
+        ConnectionDescriptor noCreds = new ConnectionDescriptor(targetId, null);
+
+        try {
+            return new IntermediateResponse<CredentialTestResult>()
+                    .body(
+                            tcm.executeConnectedTask(
+                                    noCreds,
+                                    (conn) -> {
+                                        conn.connect();
+                                        return CredentialTestResult.NA;
+                                    }));
+        } catch (Exception e1) {
+            if (AbstractAuthenticatedRequestHandler.isJmxAuthFailure(e1)) {
```

Comment:
I suspect this utility function will fail in the case of Agent HTTP connections where we have the wrong credentials, but maybe it does happen to work still. Could you test it out? I think this could be done with a procedure like:

```bash
$ sh smoketest.sh h2mem # just to make sure we have a clean database
# open web-client, let everything settle
# delete the stored credentials for both of the agents
# try this test endpoint against the agent. you won't know what the credentials are anyway, so the result should always be a `FAILURE` response.
# you could try rebuilding an agent for testing that prints out its required credentials on startup or something so you can make sure that the success path also works
```

Reply:
Anyway, this normally should not happen anyway because the Agent should always publish its stored credentials to Cryostat, particularly after https://github.com/cryostatio/cryostat-agent/pull/86 , so it's unlikely this gets hit in practice unless someone is trying to hit it.

Reply:
Is this the test fail we are talking about?

![Screenshot from 2023-03-27 21-13-01](https://user-images.githubusercontent.com/68053619/228101227-7f066b1a-02f1-4715-8be1-122353a3a256.png)

Reply:
I am seeing this:

![Screenshot from 2023-03-27 21-43-14](https://user-images.githubusercontent.com/68053619/228105070-c7145295-7353-408c-90be-0948e1df89ba.png)

Reply:
NullPointerException is surprising, but I think in general there should be some way we can detect these failures similarly to how we detect the JMX failures. It won't necessarily be 100% robust since it relies on the contents of a stack trace, but it's reasonably stable and better than just throwing raw error messages at the user.